src/fairseq2/models/conformer/convolution.py

```diff
@@ -14,6 +15,49 @@
 from fairseq2.typing import DataType, Device
 
 
+class CausalConv1d(Conv1d):
+    def __init__(
+        self,
+        in_channels: int,
+        out_channels: int,
+        kernel_size: int,
+        *,
+        stride: int = 1,
+        padding: str = "same",
+        dilation: int = 1,
+        groups: int = 1,
+        bias: bool = False,
+        padding_mode: str = "zeros",
+        device: Optional[Device] = None,
+        dtype: Optional[DataType] = None,
+    ):
+        if padding != "same":
+            raise ValueError(
+                f"We currently only support 'same' padding, and not {padding} padding."
+            )
+
+        super().__init__(
+            in_channels,
+            out_channels,
+            kernel_size,
+            stride=stride,
+            # No padding, since we are manually padding.
+            padding=0,
+            dilation=dilation,
+            groups=groups,
+            bias=bias,
+            padding_mode=padding_mode,
+            device=device,
+            dtype=dtype,
+        )
+        # Pad the last dimension entirely on the left to make the convolution causal.
+        self.pad = (kernel_size - 1, 0)
+
+    def forward(self, x: Tensor) -> Tensor:
+        x = F.pad(x, self.pad)
```

Comment:
Do we use this `CausalConv1d` anywhere else beyond a `ConformerConvolution`? If not, I don't think that it is necessary to have a module for this functionality. We can achieve the same within `ConformerConvolution`'s forward method like:

```python
def forward(...):
   ...

   if self.depthwise_conv_is_causal:
       seqs = F.pad(seqs, (self.depthwise_conv.kernel_size[0] - 1, 0))
   
   seqs = self.depthwise_conv(seqs)
```

Reply:
Also, do we have any literature reference for this padding technique?

Reply:
I believe it's used in WaveNet and similar architectures: https://www.deepmind.com/blog/wavenet-a-generative-model-for-raw-audio

It's a pretty common idea, so I wondered if we should have it as a module for anyone to leverage.

Reply:
Alrighty.